synthtool/languages/node_mono_repo.py

```diff
@@ -0,0 +1,397 @@
+
+    Returns:
+        data - package.json file as a dict.
+    """
+    with open(Path(relative_dir, "./package.json").resolve()) as f:
+        data = json.load(f)
+
+        if not all(key in data for key in _REQUIRED_FIELDS):
+            raise RuntimeError(
+                f"package.json is missing required fields {_REQUIRED_FIELDS}"
+            )
+
+        repo = git.parse_repo_url(data["repository"])
+
+        data["repository"] = f'{repo["owner"]}/{repo["name"]}'
+        data["repository_name"] = repo["name"]
+        data["lib_install_cmd"] = f'npm install {data["name"]}'
+        engines_field = re.search(r"([0-9][0-9])", data["engines"]["node"])
+        assert engines_field is not None
+        data["engine"] = engines_field.group()
+
+        return data
+
+
+def template_metadata(relative_dir: str) -> Dict[str, Any]:
+    """Load node specific template metadata.
+
+    Returns:
+        Dictionary of metadata. Includes the entire parsed contents of the package.json file if
+        present. Other expected fields:
+        * quickstart (str): Contents of the quickstart snippet if available, otherwise, ""
+        * samples (List[Dict[str, str]]): List of available samples. See synthtool.gcp.samples.all_samples()
+    """
+    metadata = {}
+    try:
+        metadata = read_metadata(relative_dir)
+    except FileNotFoundError:
+        pass
+
+    all_samples = samples.all_samples([str(Path(relative_dir, "samples/*.js"))])
+
+    # quickstart.js sample is special - only include it in the samples list if there is
+    # a quickstart snippet present in the file
+    quickstart_snippets = list(
+        snippets.all_snippets_from_file(
+            str(Path(relative_dir, "samples/quickstart.js").resolve())
+        ).values()
+    )
+    metadata["quickstart"] = quickstart_snippets[0] if quickstart_snippets else ""
+    metadata["samples"] = list(
+        filter(
+            lambda sample: sample["file"] != "samples/quickstart.js"
+            or metadata["quickstart"],
+            all_samples,
+        )
+    )
+    return metadata
+
+
+def get_publish_token(package_name: str):
+    """
+    parses the package_name into the name of the token to publish the package.
+
+    Example:
+        @google-cloud/storage => google-cloud-storage-npm-token
+        dialogflow => dialogflow-npm-token
+
+    Args:
+        package: Name of the npm package.
+    Returns:
+        The name of the key to fetch the publish token.
+    """
+    return package_name.strip("@").replace("/", "-") + "-npm-token"
+
+
+def extract_clients(filePath: Path) -> List[str]:
+    """
+    parse the client name from index.ts file
+
+    Args:
+        filePath: the path of index.ts.
+    Returns:
+        Array of client name string extract from index.ts file.
+    """
+    with open(filePath, "r") as fh:
+        content = fh.read()
+    return re.findall(r"\{(.*Client)\}", content)
+
+
+def generate_index_ts(
```

Comment:
I think several of these helpers are identical to `languages/node.py`?

I think you could import the helpers from `node.py`, so that we don't have duplicate code (you can import individual functions).